contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -6419,10 +6406,14 @@ bbf_ExecDropStmt(DropStmt *stmt)
 										 lfirst(cell),
 										 &relation,
 										 AccessShareLock,
-										 true);
+										 is_missing);
 			Assert(relation == NULL);
 			if (!OidIsValid(address.objectId))
 				continue;
+				
+			/* Restrict dropping of extended stored procedures for non-sysadmin roles */
+			if (stmt->removeType == OBJECT_PROCEDURE && !is_member_of_role(GetSessionUserId(), get_sysadmin_oid()))
```

Comment:
1. Wanted to clear out that -- 'Even w/o current code change, Only sysadmin-member were able to drop these procedures.' Is this true? If it's true then there is no point of this change.

2. Please verify whether this check blocks dropping procedures for superuser or not? Ideally it should block it.

Reply:
Correct, even without current change, only sysadmin member will be able to drop these. I'll update the logic to throw error accordingly and allow only superuser to drop it.